src/eterna/folding/EternafoldThreshknot.ts

```diff
@@ -0,0 +1,172 @@
+                    if (!(bpListFlat.includes(base1) || bpListFlat.includes(base2))) {
+                        // -1 to account for 0 indexing
+                        if (prob >= Pmax[base1 - 1]) Pmax[base1 - 1] = prob;
+                        if (prob >= Pmax[base2 - 1]) Pmax[base2 - 1] = prob;
+                    }
+                }
+
+                // Compare each COO tuple to Pmax and theta, and add to the list
+                // of selected base pairs if the check passes
+                for (let index = 0; index < bpp.length; index += 3) {
+                    const base1 = bpp[index];
+                    const base2 = bpp[index + 1];
+                    const prob = bpp[index + 2];
+                    if (prob === Pmax[base1 - 1] && prob === Pmax[base2 - 1] && prob > _theta) {
+                        current_bpList.push([base1, base2]);
+                    }
+                }
+                // Update iteration flags
+                newBP = current_bpList.length;
+                iteration += 1;
+                if (newBP !== 0 && iteration > maxIterations) {
+                    log.debug('Reached max iteration, stopping before converged.');
+                } else {
+                // Add selected base pairs to output list
+                    bpList.push(...current_bpList);
+                }
+            }
+
+            // Ensure that the bpList is sorted, then check for duplicated nucleotides
+            bpList.sort((bpA, bpB) => bpA[0] - bpB[0]);
+            const nts = bpList.flat();
+            if (nts.length > Array.from(new Set(nts)).length) {
+                log.warn('Some nucletotides found in more than 1 base pair');
+                bpList.forEach((bpA, i) => {
+                    const bpB = bpList[i + 1] || [];
+                    if (bpA[0] === bpB[0] && bpA[1] === bpB[1]) {
+                        log.warn(`Removing duplicated base pair: ${bpA}`);
+                        bpList.splice(i, 1);
+                    } else if (bpB.includes(bpA[0])) {
+                        log.warn(`bpA: ${bpA}, bpB: ${bpB}`);
+                    } else if (bpB.includes(bpA[1])) {
+                        log.warn(`bpA: ${bpA}, bpB: ${bpB}`);
+                    }
+                });
+            }
+
+            // Convert from array of pair tuples to partner pair list
+            const partnerPairList = new Array(seqStr.length).fill(-1);
+            bpList.forEach((pair) => {
+                // Indices are 1-indexed here; Dot Plot returns bp 1-indexed
+                partnerPairList[pair[0] - 1] = pair[1] - 1;
+                partnerPairList[pair[1] - 1] = pair[0] - 1;
+            });
+
+            // Convert to SecStruct and remove single pair helices
+            const structure = new SecStruct(partnerPairList);
+            const helices = structure.stems();
+            const pruned_helices = helices.filter((stem) => stem.length >= minLenHelix).flat();
```

Comment:
Can you check this PR over to make sure all variables are snake case? Our lint commit hook should've prevented this - might be a good idea to do a manual lint run in case there were other issues - did you see the commit hooks running when you comitted?

Reply:
I saw the husky commit hooks run, so I'm not sure why it didn't catch that. What's your feeling on something like `bpList` and `currentBpList`? I don't really like the `Bp` and `bp`. Maybe suffix it instead?